File: src/test/java/io/github/romeoahmed/cursedoath/combat/MeleeDamageTest.java
package io.github.romeoahmed.cursedoath.combat;

import static org.junit.jupiter.api.Assertions.*;

import java.util.List;
import org.jspecify.annotations.NullMarked;
import org.junit.jupiter.api.Test;

@NullMarked
class MeleeDamageTest {
    @Test
    void blackFlashUsesPowerGrowthAndStrengthensBareHands() {
        assertEquals(32f, MeleeDamage.resolve(4f, true));
        assertEquals(129.64181, MeleeDamage.resolve(7f, true), 0.0001);
        assertEquals(5.656854f, MeleeDamage.resolve(1f, true), 0.000001f);
    }

    @Test
    void ordinaryReinforcementAndZeroDamageStaySeparateFromBlackFlash() {
        assertEquals(4.6f, MeleeDamage.resolve(4f, false));
        for (var blackFlash : List.of(false, true)) {
            assertEquals(0f, MeleeDamage.resolve(0f, blackFlash));
            assertEquals(0f, MeleeDamage.resolve(-1f, blackFlash));
        }
    }
}

File: src/test/kotlin/io/github/romeoahmed/cursedoath/combat/MeleeDamageTest.kt
package io.github.romeoahmed.cursedoath.combat

import kotlin.test.Test
import kotlin.test.assertEquals

class MeleeDamageTest {
    @Test
    fun `black flash uses power growth and strengthens bare hands`() {
        assertEquals(32f, MeleeDamage.resolve(4f, true))
        assertEquals(129.64181f, MeleeDamage.resolve(7f, true), 0.0001f)
        assertEquals(5.656854f, MeleeDamage.resolve(1f, true), 0.000001f)
    }

    @Test
    fun `ordinary reinforcement and zero damage stay separate from black flash`() {
        assertEquals(4.6f, MeleeDamage.resolve(4f, false))
        for (blackFlash in listOf(false, true)) {
            assertEquals(0f, MeleeDamage.resolve(0f, blackFlash))
            assertEquals(0f, MeleeDamage.resolve(-1f, blackFlash))
        }
    }
}
